pysph/base/opencl.py

```diff
@@ -35,6 +35,47 @@ def set_queue(q):
     _queue = q
 
 
+class DeviceArray(cl.array.Array):
+    def __init__(self, queue, dtype, n=0):
+        self.queue = queue
```

Comment:
Don't need this, just use `get_queue` to get a suitable queue.  I don't know if we need multiple separate queues?